### PR TITLE
TEST/UCP/WIREUP: Fix and enhance wireup test

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6682/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6684/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov @brminich pls take a look

File: test/gtest/ucp/test_ucp_wireup.cc [hunk 1]
```diff
@@ -56,50 +62,76 @@ class test_ucp_wireup : public ucp_test {
...
     void clear_recv_data();
 
-    ucp_rkey_h get_rkey(ucp_mem_h memh);
+    ucp_rkey_h get_rkey(ucp_ep_h ep, ucp_mem_h memh);
 
     static void send_completion(void *request, ucs_status_t status);
 
-    static void recv_completion(void *request, ucs_status_t status,
+    static void tag_recv_completion(void *request, ucs_status_t status,
                                 ucp_tag_recv_info_t *info);
```
Comment (brminich) [inline test/gtest/ucp/test_ucp_wireup.cc]: alignment

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6695/ for details (Mellanox internal link).

**Pull request merged**